tools/flang2/flang2exe/llutil.cpp

```diff
@@ -2198,77 +2198,70 @@ metadata_args_need_struct(void)
  * This function returns true for the types supported
  * in function make_param_op
  */
-bool
-should_preserve_param(SPTR sptr)
-{
-  bool should_preserve = false;
+bool should_preserve_param(SPTR sptr) {
   DTYPE dtype = DTYPEG(sptr);
 
   switch (DTY(dtype)) {
-    case TY_ARRAY:
-    case TY_STRUCT:
-    case TY_BLOG:
-    case TY_SLOG:
-    case TY_LOG:
-    case TY_BINT:
-    case TY_SINT:
-    case TY_INT:
-    case TY_REAL:
-    case TY_INT8:
-    case TY_LOG8:
-    case TY_DBLE:
-    case TY_QUAD:
-    case TY_CMPLX:
-    case TY_DCMPLX:
-    case TY_CHAR:
-      should_preserve = true;
-      break;
+  case TY_ARRAY:
+  case TY_STRUCT:
+  case TY_BLOG:
+  case TY_SLOG:
+  case TY_LOG:
+  case TY_BINT:
+  case TY_SINT:
+  case TY_INT:
+  case TY_REAL:
+  case TY_INT8:
+  case TY_LOG8:
+  case TY_DBLE:
+  case TY_QUAD:
+  case TY_CMPLX:
+  case TY_DCMPLX:
+  case TY_CHAR:
+    return true;
+  default:
+    return false;
   }
-  return should_preserve;
 }
 
-OPERAND *
-make_param_op(SPTR sptr)
-{
+OPERAND *make_param_op(SPTR sptr) {
   OPERAND *oper;
   DTYPE dtype = DTYPEG(sptr);
 
-  assert(should_preserve_param(sptr), "make_param_op(): unexpected sptr",
-         0, ERR_Fatal);
+  assert(should_preserve_param(sptr), "make_param_op(): unexpected sptr", 0,
```

Comment:
> Yes I agree that it looks a little odd. But it has to be done to identify the (now) unsupported cases and let it generate an IR without (out of scope) parameter debug-info in place of producing an error which it would have done without this PR.


You can add unsupported cases in this switch statement qualifying them with a comment which mentions that they are unsupported. These cases need not generate an error. The default case can safely generate errors how any type which are not expected to be handled.

```
switch(DTY(dtype)) {
// handled cases
case TY_BLOG:
.
.
// unsupported cases
case TY_<foo>:
case TY_<foo2>:
    break;
default:
   <error message>
```

Reply:
That looks great. I shall incorporate that.